apis/metal3.io/v1alpha1/hostfirmwarecomponents_types.go

```diff
@@ -0,0 +1,118 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// FirmwareUpdate defines a firmware update specification.
+type FirmwareUpdate struct {
+	Name string `json:"name"`
```

Comment:
This is the component name, right? Let's call it `Component` then (sorry, did not spot the potential inconsistency on the spec).

Reply:
correct it's the component name, I will update it.